### DISCUSSION TITLE
How to make a button appear only when input is focused

### DISCUSSION THREAD

**Discussion by 549393092** [quasarframework/quasar#11849] [General - Components / Directives / etc] [answered]
Dear Quasar team,

I did some research about this topic. see below link
https://stackoverflow.com/questions/57930783/how-to-make-a-button-appear-only-when-input-is-focused
However, as you know, with standard q-input, we will embed q-btn into q-input (not sibling in the above link). I would like to know how to implement it.
I also did a demo to illustrate it.
see link https://codepen.io/Stanley-549393092/pen/qBPxyGK?editors=1111
My purpose is to show more text in q-input, so as you can see in the demo, it can only display 123, and I would like to show more.

Thanks for your help.

**Comment by MilosPaunovic** [comment, marked as answer]
[https://codepen.io/milospaunovic/pen/vYedzRY?editors=1111](https://codepen.io/milospaunovic/pen/vYedzRY?editors=1111)

**Reply by 549393092** [reply]
@MilosPaunovic Thank you! But I found if the button appears in the first line, the second line will appear as well. Also it may not be a generic way, for example, if there a lot multiple q-input, it has to define multiples variables.

**Reply by MilosPaunovic** [reply]
There is really no, as you say, generic way to accomplish that.
I may suggest another solution, to create your own wrapper component for `QSelect`, and handle `@focus` & `@blur` events in it.

**Reply by smolinari** [reply]
This works. https://codepen.io/smolinari/pen/ZEXrmPe?editors=1011

Scott

**Reply by 549393092** [reply]
Thanks @MilosPaunovic @smolinari 
I will work on this direction. But now it faces another issue, when q-input is in read only mode, the focus event will not be triggered. I would like to show the button even if q-input is in read only mode. Do you have any idea?

**Reply by smolinari** [reply]
@549393092 - If the field is read only, it should be immutable. So, there is no ability to give it focus to make a change. Or put another way, read only means the user isn't allowed to make changes and shouldn't be offered any way to make changes. I'd say the requirement is a bit "off".  

Scott

**Reply by 549393092** [reply]
Hi Scott,
Actually this is my requirement. In read only (display) mode, the user may still want to see the possible values from value help button. Or else it has to switch to edit mode every time. Of course, the program has to control that user can't update anything in display mode.
Thanks!

**Reply by smolinari** [reply]
@549393092 - Hmmm. Ok. How about this? 

https://codepen.io/smolinari/pen/ZEXrmPe?editors=1011

Scott

**Reply by 549393092** [reply]
@smolinari Thanks a lot for this great idea. Although it might be a little difficult to encapsulate the logic.
Last question, I am thinking about why focus event is not triggered with q-input in read only mode. 
I tested native input, it will trigger focus event. Could you please see below demo?
https://codepen.io/Stanley-549393092/pen/XWeEbaw?editors=1111

**Reply by smolinari** [reply]
@549393092 - I'm not enough of a frontend dev (or a Quasar dev, despite my maintainer tag) to give you the low down as to why the focus event doesn't fire with the readonly mode. I can only imagine it is because the logic seems to be the field isn't allowing input, so focusing and blurring isn't necessary in most use cases. 

What do you mean by encapsulating the logic?

Scott

**Reply by 549393092** [reply]
@smolinari Got it. I mean it's a little difficult to wrapper the q-input. For example, if q-input is embeded in table, it has to implement mouseover and mouseleave event with tag q-tr, and if q-input is an individual element (without in table), it has to find another tag to implement mouseover and mouseleave event.

**Reply by smolinari** [reply]
@549393092 - Encapsulated!

https://codepen.io/smolinari/pen/jOGzbKE?editors=1111

Scott

**Reply by 549393092** [reply]
Hi Scott,

Wow, it's fantastic. Thanks a lot for your support!
